pkg/virtctl/imageupload/pvc.go

```diff
@@ -0,0 +1,121 @@
+package imageupload
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+
+	storagetypes "kubevirt.io/kubevirt/pkg/storage/types"
+)
+
+func (c *command) createUploadPVC() (*v1.PersistentVolumeClaim, error) {
+	if c.accessMode == string(v1.ReadOnlyMany) {
+		return nil, fmt.Errorf("cannot upload to a readonly volume, use either ReadWriteOnce or ReadWriteMany if supported")
+	}
+
+	// We check if the user-defined storageClass exists before attempting to create the PVC
+	if c.storageClass != "" {
+		_, err := c.client.StorageV1().StorageClasses().Get(context.Background(), c.storageClass, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	quantity, err := resource.ParseQuantity(c.size)
+	if err != nil {
+		return nil, fmt.Errorf("validation failed for size=%s: %s", c.size, err)
+	}
+	pvc := storagetypes.RenderPVC(&quantity, c.name, c.namespace, c.storageClass, c.accessMode, c.volumeMode == "block")
+	if c.volumeMode == "filesystem" {
+		volMode := v1.PersistentVolumeFilesystem
+		pvc.Spec.VolumeMode = &volMode
+	}
+
+	contentType := string(cdiv1.DataVolumeKubeVirt)
+	if c.archiveUpload {
+		contentType = string(cdiv1.DataVolumeArchive)
+	}
+
+	annotations := map[string]string{
+		uploadRequestAnnotation: "",
+		contentTypeAnnotation:   contentType,
+	}
+
+	if c.forceBind {
+		annotations[forceImmediateBindingAnnotation] = ""
+	}
+
+	pvc.ObjectMeta.Annotations = annotations
+	c.setDefaultInstancetypeLabels(&pvc.ObjectMeta)
+
+	pvc, err = c.client.CoreV1().PersistentVolumeClaims(c.namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return pvc, nil
+}
+
+func (c *command) ensurePVCSupportsUpload(pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+	var err error
+	_, hasAnnotation := pvc.Annotations[uploadRequestAnnotation]
+
+	if !hasAnnotation {
+		if pvc.GetAnnotations() == nil {
+			pvc.SetAnnotations(make(map[string]string, 0))
+		}
+		pvc.Annotations[uploadRequestAnnotation] = ""
+		pvc, err = c.client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(context.Background(), pvc, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return pvc, nil
+}
+
+func (c *command) getAndValidateUploadPVC() (*v1.PersistentVolumeClaim, error) {
+	pvc, err := c.client.CoreV1().PersistentVolumeClaims(c.namespace).Get(context.Background(), c.name, metav1.GetOptions{})
+	if err != nil {
+		c.cmd.Printf("PVC %s/%s not found \n", c.namespace, c.name)
```

Comment:
This is not accurate, since IsNotFound errro can be one of multiple error types  the Get function returns